### ISSUE TITLE
request to support Heat Pump Room Thermostat

### ISSUE THREAD

**Issue by Uaeguy** [make-all/tuya-local#33]
Hello,

Thank you for the time invested in creating this component. 

I have tried the add on and while adding the device it only promote the switch.

I have heat pump thermostat from company called BECA. There is custom firmware available but I have not tried it.

Having said that, Can you please check the possibilities to support this device in your addon?

[here is a link of the device it self](https://www.becasmart.com/product/beca-bhp-6000-wifi-voice-control-conventional-heat-pump-room-thermostat/)

My device generated the following 

{
  "result": {
    "active_time": 1596728546,
    "biz_type": 18,
    "category": "wk",
    "create_time": 1595768698,
    "icon": "smart/program_category_icon/wk.png",
    "id": "quoted for security",
    "ip": "quoted for security",
    "lat": "quoted for security",
    "local_key": "quoted for security",
    "lon": "quoted for security",
    "model": "热泵温控器",
    "name": "Room thermostat",
    "online": true,
    "owner_id": "quoted for security",
    "product_id": "quoted for security",
    "product_name": "HeatPump Thermostat",
    "status": [
      {
        "code": "switch",
        "value": true
      },
      {
        "code": "temp_set_f",
        "value": 77
      },
      {
        "code": "temp_current_f",
        "value": 78
      },
      {
        "code": "level",
        "value": "1"
      },
      {
        "code": "eco",
        "value": false
      },
      {
        "code": "child_lock",
        "value": false
      }
    ],
    "sub": false,
    "time_zone": "+00:00",
    "uid": "xxxxxxxxxx",
    "update_time": xxxxxx,
    "uuid": "xxxxxxxxx"
  },
  "success": true,
  "t": xxxxxxxxxx
}

if any data required please let me know and I will try to provide every thing possible to me

Thanks

**Comment by make-all** [comment]
Please send the logs messages from Home Assistant that are output when it is detected as "Simple Switch".
That will have the list of DPS ids returned by the local protocol, which we can try to match up with the functions (switch, temp_set, temp_current, level, eco, child_lock).

**Comment by make-all** [comment]
I'm thinking of using the product_id in future, so if you can uncloak that (it is linked to the product, not unique to your specific device), it might help with future plans for improving detection.

**Comment by Uaeguy** [comment]
Hello again,

Here is the log generated from the component

2021-07-22 16:00:04 INFO (MainThread) [custom_components.tuya_local.device] Setting protocol version for Test to 3.3.
2021-07-22 16:00:05 WARNING (MainThread) [custom_components.tuya_local.config_flow] Device matches simple_switch with quality of 14%. DPS: {'1': True, '2': 77, '3': 87, '4': '3', '5': '3', '6': False, '7': False, 'updated_at': 1626955204.1256018}
2021-07-22 16:00:05 WARNING (MainThread) [custom_components.tuya_local.config_flow] Report this to https://github.com/make-all/tuya-local/issues/
2021-07-22 16:00:08 INFO (MainThread) [custom_components.tuya_local.device] Creating device: "quoted for security"
2021-07-22 16:00:08 INFO (MainThread) [custom_components.tuya_local.device] Setting protocol version for Simple Switch to 3.3.


And for the product ID as requested

    "product_id": "drwezxukumr9ohuh",

**Comment by make-all** [comment]
It looks like most of the functions available on the control panel are not exposed via Wi-Fi. Only the functions mentioned in the web API capture you posted first, and one additional numeric field.  I see the temperature units are selectable, so I hope the extra numeric field is an indication of those.  But the current value is 3, and the units only seem to be selectable between C and F, so perhaps not. If the temperature units are not indicated by the response from the device, it might be more complex to support.

I have checked in an experimental config, please reinstall the Tuya Local integration using the "main" branch, which you can do from HACS using the menu button on the Tuya Local card, and selecting "Reinstall", then on the version, instead of the default latest release of 0.8.9, select main, which should be at the bottom of the list.

This hopefully will let you use the Thermostat card, and in Developer Tools you will be able to observe the rest of the attributes.  Most of interest to me is whether my guess about the C/F selection is correct, as getting this wrong will probably make some strange behavior.  Also, there are a number of attributes that might be swapped - current and target temperature, the child lock and eco modes, and the level and other numeric indicator.

**Comment by Uaeguy** [comment]
Hello again,

I will reinstall now, 

I have observed the DPS and I found the following

DPS: 
{
'1': True, "back light switch bool"


'2': 77, "set temp in Fahrenheit"



'3': 87, "ambient temp in Fahrenheit"


 '4': '3', "function, using schdule 1, temporary hold 2, permanent hold 3, holiday hold 4"
 '5': '3', "work mode, cool 1, heat 2, off 3, emergency heat 4, auto 5"



 '6': False, "fan mode bool (on = true)(auto = false)"



 '7': False, "child lock bool"

**Comment by Uaeguy** [comment]
After reinstalling, I'm able to use temperature card, I can change temperature but it is not behaving correctly. When I choose eco, the fan change from auto to on, and normal it will change to on. And when I change operation from heat/cool to off  the backlight on the device change from on to off and vice versa. 

Thank you

**Comment by Uaeguy** [comment]
If this helps the following in developer tool

also a note, I have contacted Tuya about changing from F to C and they said it is not possible to change it in app only on the physical device. So I don't think there will be DPS for that I'm afraid.

`hvac_modes:
  - heat_cool
  - 'off'
min_temp: 7
max_temp: 35
target_temp_step: 1
preset_modes:
  - normal
  - eco
current_temperature: 80
temperature: 74
preset_mode: normal
level: 3
friendly_name: Beca BHP-6000 Thermostat
supported_features: 17
`

**Comment by make-all** [comment]
If you reinstall again, I have reorganised based on your first comment from yesterday.  I created two separate configs for F and C, so you'll need to select the _F config for your device, which is configured for Fahrenheit.  Configuration from the previous version should automatically choose the Fahrenheit version, but if it doesn't, delete the device and set it up again.

**Comment by Uaeguy** [comment]
Now it is working like a charm, just like expected. Thanks a lot. 

As a side note, I have checked for C as well, when I set the value "say 20 C" it will promote the correct value and it will set it as 20 in the thermostat, however after some time in the card it will change back to the F values (68) without setting them to that value in the device which is good. I'm not sure if I'm clear. Also it is the same in the ambient temp.

Other note, for the backlight when it is switched off it will close the whole unit and I think what you have done to keep it as a separate entity same as the child lock is perfect for it Thanks and appreciated

Cheers
